pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -145,6 +150,39 @@ func (m *managerImpl) IsUnderInodePressure() bool {
 	return hasNodeCondition(m.nodeConditions, api.NodeInodePressure)
 }
 
+func startMemoryThresholdNotifier(thresholds []Threshold, observations signalObservations, hard bool, handler thresholdNotifierHandlerFunc) error {
+	for _, threshold := range thresholds {
+		if threshold.Signal != SignalMemoryAvailable ||
+			(hard && threshold.GracePeriod != time.Duration(0)) ||
```

Comment:
AFAIK, we don't support eviction grace period for `hard` thresholds. We only support `termination grace period`. cc @derekwaynecarr